### PR TITLE
Add DBFetcher generate reports ability

### PR DESCRIPTION
**Pull request by MoonBasedKas**
Updated DBFetcher to now have two new models (ReportData and ReportDataSales) which are responsible for holding onto report information. DBFetcher now is capable of making requests to the database in order to get the required information to generate reports. The two new methods that allow this are the following:

- GenerateStaleReport(String, int): ReportData
  - This will generate a stale donuts report based on a given starting date and go back x many days.
- GenerateSalesReport(String, int): ReportDataSales
  - This will generate a report for the total sales starting from a given date and go back a specified range.

### CODE REVIEW

File: Core/src/main/java/io/htmlcss/model/ReportDataSales.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package io.htmlcss.model;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.Hashtable;
+
+public class ReportDataSales extends ReportData{
+	private Dictionary<Integer, Float> dPrice = null;
```
Comment (Korbrent) [inline Core/src/main/java/io/htmlcss/model/ReportDataSales.java:7]: Doesn't make much sense to extend a class & keep a similar private variable. Maybe make the super class `protected`?

File: Core/src/main/java/io/htmlcss/model/ReportDataSales.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	
+	/**
+	 * Constructor
+	 */
+	public ReportDataSales() {
+		super();
+		this.dPrice = new Hashtable<Integer, Float>();
+	}
+
+	public void addItemPrice(int id, float price) {
```
Comment (Korbrent) [inline Core/src/main/java/io/htmlcss/model/ReportDataSales.java:17]: The same function exists in super

**Review by Korbrent — DISMISSED** [review]
Left comments on code. Functionality of super() is implemented in child without extending functionality

**Review by HeartAttackCat — APPROVED** [review]
Creates two new models. The first report model is used for stale donut tracking. This allows the front end to pull the stale donuts. The second report model is used for sales of the donuts. 

- Does not drop database

**Pull request merged**